crates/rattler_installs_packages/src/wheel_builder.rs

```diff
@@ -0,0 +1,370 @@
+        let extra_requirements = HashSet::<Requirement>::from_iter(
+            extra_requirements
+                .iter()
+                .map(|s| Requirement::from_str(s).expect("...")),
+        );
+        let combined_requirements = requirements
+            .union(&extra_requirements)
+            .cloned()
+            .collect::<Vec<_>>();
+
+        if !extra_requirements.is_empty() && requirements.len() != combined_requirements.len() {
+            let locked_packages = HashMap::default();
+            // Todo: use the previous resolve for the favored packages?
+            let favored_packages = HashMap::default();
+            let all_requirements = combined_requirements.to_vec();
+            let extra_resolved_wheels = resolve(
+                self.package_db,
+                all_requirements.iter(),
+                self.env_markers,
+                self.wheel_tags,
+                locked_packages,
+                favored_packages,
+                &self.resolve_options,
+            )
+            .await
+            .map_err(|_| WheelBuildError::CouldNotResolveEnvironment(all_requirements))?;
+
+            // install extra wheels
+            for package_info in extra_resolved_wheels {
+                if resolved_wheels.contains(&package_info) {
+                    continue;
+                }
+                tracing::info!(
+                    "Wheel building: installing extra requirements: {} - {}",
+                    package_info.name,
+                    package_info.version
+                );
+                let artifact_info = package_info.artifacts.first().unwrap();
+                let artifact = self
+                    .package_db
+                    .get_artifact::<Wheel>(artifact_info)
+                    .await
+                    .expect("Could not get artifact");
+
+                cache_value.venv.install_wheel(&artifact, &options)?;
+            }
+        }
+
+        self.venv_cache
+            .borrow_mut()
+            .insert(sdist.name().clone(), Rc::new(cache_value));
+
+        return self
+            .venv_cache
+            .borrow()
+            .get(sdist.name())
+            .cloned()
+            .ok_or_else(|| WheelBuildError::Error("Could not get venv from cache".to_string()));
+    }
+
+    fn run_command(&self, cache: &CacheValue, stage: &str) -> std::io::Result<Output> {
+        // three args: cache.folder, goal
+        Command::new(cache.venv.python_executable())
+            .current_dir(&cache.package_dir)
+            .arg(cache.folder.path().join("build_frontend.py"))
+            .arg(cache.folder.path())
+            .arg(&cache.entry_point)
+            .arg(stage)
+            .output()
+    }
+
+    /// Get the metadata for a given sdist by using the build_backend in a virtual env
+    /// This function uses the `prepare_metadata_for_build_wheel` entry point of the build backend.
+    pub async fn get_metadata(
```

Comment:
I think we do not need to save the PackageDB in this struct directly right? We could also pass it to the `get_metadata` and `build_wheel` method. Maybe that would simplify things?

Reply:
true. the current way seems to work fine, but I guess we could also pass the packagedb to the functions ... 
For the venv caching to work it might make sense to be sure that the packagedb is the same for both functions so maybe that would be an argument for keeping it in the struct.